src/components/course/data/hooks.jsx

```diff
@@ -905,3 +907,41 @@ export const useMinimalCourseMetadata = () => {
   };
   return courseMetadata;
 };
+
+export const useExternalEnrollmentFailureReason = () => {
+  const intl = useIntl();
+  const {
+    userSubsidyApplicableToCourse,
+    missingUserSubsidyReason,
+  } = useContext(CourseContext);
+  return useMemo(() => {
+    if (userSubsidyApplicableToCourse) {
+      return {};
+    }
+    let reason;
+    const noSubsidyReasons = [
+      DISABLED_ENROLL_REASON_TYPES.NO_SUBSIDY_NO_ADMINS,
+      DISABLED_ENROLL_REASON_TYPES.NO_SUBSIDY,
+      DISABLED_ENROLL_REASON_TYPES.POLICY_NOT_ACTIVE,
+    ];
+    const systemErrorReasons = [
+      DISABLED_ENROLL_REASON_TYPES.CONTENT_NOT_IN_CATALOG,
+      DISABLED_ENROLL_REASON_TYPES.LEARNER_NOT_IN_ENTERPRISE,
+    ];
+    if (!missingUserSubsidyReason || noSubsidyReasons.includes(missingUserSubsidyReason?.reason)) {
+      reason = 'no_offer_available';
```

Comment:
Should we put these magic strings in a constant, like we've done [here](
https://github.com/openedx/frontend-app-learner-portal-enterprise/blob/42d4d78b1f8cb3e041e4e3bcf53e8bad3f3ddbe5/src/components/course/data/constants.js#LL75C23-L75C23) ?

Reply:
Probably; that said, these magic strings are in the previous, now-deprecated flow. I'm leaning more on the side of leave them as is until we remove the deprecated components in favor of only the new flow instead, and put them in a constant at that point. Thoughts?

Reply:
Sounds good!